src/Microsoft.Diagnostics.Monitoring.RestServer/Controllers/DiagController.cs

```diff
@@ -79,20 +82,67 @@ public Task<ActionResult> GetGcDump(int? pid)
         public Task<ActionResult> Trace(
             int? pid,
             [FromQuery]TraceProfile profile = DefaultTraceProfiles,
-            [FromQuery][Range(-1, int.MaxValue)]int durationSeconds = 30,
+            [FromQuery][Range(-1, int.MaxValue)] int durationSeconds = 30,
             [FromQuery][Range(1, int.MaxValue)] int metricsIntervalSeconds = 1)
         {
             TimeSpan duration = ConvertSecondsToTimeSpan(durationSeconds);
-            return InvokeService(async () =>
+
+            var configurations = new List<MonitoringSourceConfiguration>();
+            if (profile.HasFlag(TraceProfile.Cpu))
             {
-                int pidValue = _diagnosticServices.ResolveProcess(pid);
-                IStreamWithCleanup result = await _diagnosticServices.StartTrace(pidValue, profile, duration, metricsIntervalSeconds, this.HttpContext.RequestAborted);
-                return new StreamWithCleanupResult(result, "application/octet-stream", FormattableString.Invariant($"{Guid.NewGuid()}.nettrace"));
-            });
+                configurations.Add(new CpuProfileConfiguration());
+            }
+            if (profile.HasFlag(TraceProfile.Http))
+            {
+                configurations.Add(new HttpRequestSourceConfiguration());
+            }
+            if (profile.HasFlag(TraceProfile.Logs))
+            {
+                configurations.Add(new LoggingSourceConfiguration());
+            }
+            if (profile.HasFlag(TraceProfile.Metrics))
+            {
+                configurations.Add(new MetricSourceConfiguration(metricsIntervalSeconds));
+            }
+
+            var aggregateConfiguration = new AggregateSourceConfiguration(configurations.ToArray());
+
+            return StartTrace(pid, aggregateConfiguration, duration);
+        }
+
+        [HttpPost("trace/{pid?}")]
+        public Task<ActionResult> TracePostedConfiguration(
```

Comment:
nit: TraceCustomConfiguration? The verb isn't really important, the fact that it's user specified is.